indigo/indigo/src/main/scala/indigo/shared/events/GlobalEvent.scala

```diff
@@ -189,16 +173,89 @@ object MouseEvent:
     * @param amount
     *   vertical amount of pixels, pages or other unit, depending on delta mode, the Y axis was scrolled
     */
-  final case class Wheel(position: Point, amount: Double) extends MouseEvent:
-    val x: Int = position.x
-    val y: Int = position.y
+  final case class Wheel(position: Point, amount: Double) extends MouseEvent
 
   object Wheel:
     def apply(x: Int, y: Int, amount: Double): Wheel =
       Wheel(Point(x, y), amount)
 
 end MouseEvent
 
+/** Represents all mouse, pen and touch events
+  */
+sealed trait PointerEvent extends InputEvent:
+  import PointerEvent.*
+
+  /** Coordinates relative to the magnification level
+    */
+  def position: Point
+
+  /** Unique pointer identifier
+    */
+  def pointerId: PointerId
+
+  /** Pressed buttons
+    */
+  def buttons: Buttons
+
+  /** Indicates whether buttons are in active state
+    */
+  def isActive: Boolean = buttons != Buttons(0)
+
+  /** Indicates whether the pointer is considered primary - like first finger during multi-touch gesture
+    */
+  def isPrimary: Boolean
+
+object PointerEvent:
+  /** Unique pointer identifier. Could be used to distinguish between pointers in multi-touch interactions
+    */
+  opaque type PointerId = Double
+  object PointerId:
+    inline def apply(id: Double): PointerId = id
+
+    given CanEqual[PointerId, PointerId] = CanEqual.derived
+
+  /** A number representing one or more buttons: https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/buttons
+    */
+  opaque type Buttons = Int
+  object Buttons:
+    inline def apply(buttons: Int): Buttons = buttons
+
+    given CanEqual[Buttons, Buttons] = CanEqual.derived
+
+  /** Pointing device is moved into canvas hit test boundaries. It's counterpart is [[PointerLeave]].
+    */
+  case class PointerEnter(position: Point, pointerId: PointerId, buttons: Buttons, isPrimary: Boolean)
```

Comment:
All case classes should be `final` please. These's a bunch here and also one in the demo scene at the bottom.